internal/repository/sync_test.go

```diff
@@ -218,23 +189,59 @@ func TestClone(t *testing.T) {
 		}
 		defer os.Chdir(currentDir) //nolint:errcheck
 
-		repo, err := Clone(context.Background(), remoteTmpDir+"//", "", "")
+		repo, err := Clone(context.Background(), remoteTmpDir+"//", "", "", nil)
 		assert.Nil(t, err)
+
 		head, err := repo.r.Head()
 		if err != nil {
 			t.Fatal(err)
 		}
 		assert.Equal(t, commitID, head.Hash())
 
-		localRSLRef, err := repo.r.Reference(plumbing.ReferenceName(rsl.Ref), true)
+		assertLocalAndRemoteRefsMatch(t, repo.r, remoteRepo.r, rsl.Ref)
+		assertLocalAndRemoteRefsMatch(t, repo.r, remoteRepo.r, policy.PolicyRef)
+	})
+
+	t.Run("successful clone without specifying dir, with expected root keys", func(t *testing.T) {
+		localTmpDir := t.TempDir()
+
+		if err := os.Chdir(localTmpDir); err != nil {
+			t.Fatal(err)
+		}
+		defer os.Chdir(currentDir) //nolint:errcheck
+
+		publicKey, err := tuf.LoadKeyFromBytes(rootKeyBytes)
 		if err != nil {
 			t.Fatal(err)
 		}
-		assert.Equal(t, remoteRSLRef.Hash(), localRSLRef.Hash())
-		localPolicyRef, err := repo.r.Reference(plumbing.ReferenceName(policy.PolicyRef), true)
+
+		repo, err := Clone(context.Background(), remoteTmpDir, "", "", []*tuf.Key{publicKey})
```

Comment:
could you add a test with multiple root keys?

Reply:
I have replaced the single root key tests with multiple root key tests.